spec/cerner/oauth1a/access_token_agent_spec.rb

```diff
@@ -3,12 +3,82 @@
 require 'spec_helper'
 require 'mock_access_token_server'
 
+require 'base64'
 require 'cerner/oauth1a/access_token_agent'
 require 'cerner/oauth1a/access_token'
 require 'cerner/oauth1a/oauth_error'
+require 'json'
 
 # rubocop:disable Metrics/BlockLength
 RSpec.describe Cerner::OAuth1a::AccessTokenAgent do
+  describe '#retrieve_keys' do
+    before(:all) do
+      @server = MockAccessTokenServer.new(
+        [
+          {
+            path: '/oauth/access', response: {
+              status: 200,
+              content_type: 'application/x-www-form-urlencoded',
+              body: 'oauth_token=TOKEN'\
+                '&oauth_token_secret=TOKEN%20SECRET'\
+                '&oauth_session_handle=SESSION'\
+                '&oauth_expires_in=3600'\
+                '&oauth_authorization_expires_in=86400'
+            }
+          },
+          {
+            path: '/oauth/access/keys/1', response: {
+              status: 200,
+              content_type: 'application/json',
+              body: JSON.generate(
+                aesKey: {
+                  secretKey: Base64.encode64('123456')
+                },
+                rsaKey: {
+                  publicKey: Base64.encode64('789012')
+                }
+              )
+            }
+          },
+          {
+            path: '/oauth/access/keys/token_rejected', response: {
+              status: 401,
+              content_type: 'text/plain',
+              www_authenticate: 'OAuth realm="http%3A%2F%2Flocalhost", oauth_problem="token_rejected"',
+              body: 'TOKEN REJECTED'
+            }
+          }
+        ]
+      )
+      @server.startup
+    end
+
+    after(:all) do
+      @server.shutdown
+    end
+
+    it 'gets valid keys' do
+      agent = Cerner::OAuth1a::AccessTokenAgent.new(
+        access_token_url: "#{@server.base_uri}/oauth/access",
+        consumer_key: 'CONSUMER KEY',
+        consumer_secret: 'CONSUMER SECRET'
+      )
+      keys = agent.retrieve_keys('1')
+      expect(keys.version).to eq('1')
+      expect(keys.aes_secret_key).to eq('123456')
+      expect(keys.rsa_public_key).to eq('789012')
+    end
+
+    it 'throw OAuthError with token_rejected oauth_problem' do
```

Comment:
Minor, but can we use contexts for the specs to give a little more information about what is being tested (i.e. a valid key version above and an invalid key version below). Also I feel like we are missing specs for when the key version is nil.

Reply:
Yeah, the nil param test is missing.